scan

```diff
@@ -91,74 +129,357 @@ def run(options=None):
 #
 # Produces a CSV for each scan, with each domain and results.
 ###
-def scan_domains(scanners, domains):
-
+def scan_domains(scanners, domains, options):
     # Clear out existing result CSVs, to avoid inconsistent data.
     for result in glob.glob("%s/*.csv" % utils.results_dir()):
         os.remove(result)
 
+    # Trigger the Lambda pipeline.
+    use_lambda = options.get("lambda", False)
+
+    # Store local errors/timing info, and if using Lambda, trigger the
+    # Lambda post-processing pipeline to get Lambda timing/usage info.
+    meta = options.get("meta", False)
+
     # Run through each scanner and open a file and CSV for each.
     handles = {}
     for scanner in scanners:
         name = scanner.__name__.split(".")[-1]  # e.g. 'pshtt'
         scanner_filename = "%s/%s.csv" % (utils.results_dir(), name)
         scanner_file = open(scanner_filename, 'w', newline='')
         scanner_writer = csv.writer(scanner_file)
-        scanner_writer.writerow(["Domain", "Base Domain"] + scanner.headers)
+
+        # Write the header row, factoring in Lambda detail if needed.
+        headers = prefix_headers + scanner.headers
+
+        # Local scan timing/errors.
+        if meta:
+            headers += local_headers
+
+        # Lambda scan timing/errors. (At this step, only partial fields.)
+        if meta and use_lambda:
+            headers += lambda_headers
+
+        scanner_writer.writerow(headers)
 
         handles[scanner] = {
+            'name': name,
             'file': scanner_file,
             'filename': scanner_filename,
-            'writer': scanner_writer
+            'writer': scanner_writer,
+            'headers': headers
         }
 
-    # The task wrapper that's parallelized using executor.map.
-    def process_scan(params):
-        scanner, domain, options = params
-
-        # A scanner can return multiple rows.
-        try:
-            rows = list(scanner.scan(domain, options))
-        except:
-            logging.warn(utils.format_last_exception())
+    # Experimental!
+    if use_lambda:
+        scan_method = perform_lambda_scan
+    else:
+        scan_method = perform_local_scan
```

Comment:
This is where things branch for Lambda v local, by assigning a different method to the method that gets sent to each worker. However, eventually there became a lot of copy/pasted code between those two methods. So a TODO is to branch this deeper in by sending the same method to all workers, and branching within that method.